packages/markdown-it-myst/tests/roles.spec.ts

```diff
@@ -22,23 +22,40 @@ describe('parses roles', () => {
     expect(tokens[1].children?.[1].content).toEqual('hello');
     expect(tokens[1].children?.[3].content).toEqual('hello');
   });
-  it('header role parses', () => {
+  it('basic role is inline', () => {
+    const mdit = MarkdownIt().use(plugin);
+    const tokens = mdit.parse('ok {sup}`+`', {});
+    expect(tokens.map((t) => t.type)).toEqual(['paragraph_open', 'inline', 'paragraph_close']);
+    expect(tokens[1].children?.map((t) => t.type)).toEqual([
+      'text',
+      'parsed_role_open',
+      'role_body_open',
+      'inline',
+      'role_body_close',
+      'parsed_role_close',
+    ]);
+    expect(tokens[1].content).toEqual('ok {sup}`+`');
+    // Pass the column information for the role
+    expect((tokens[1].children?.[1] as any).col).toEqual([3, 11]);
+    expect(tokens[1].children?.[1].info).toEqual('sup');
+    expect(tokens[1].children?.[1].content).toEqual('+');
+    expect(tokens[1].children?.[3].content).toEqual('+');
+  });
+  it('role is inline (no headings)', () => {
     const mdit = MarkdownIt().use(plugin);
     const tokens = mdit.parse('{abc}`# hello`', {});
     expect(tokens.map((t) => t.type)).toEqual(['paragraph_open', 'inline', 'paragraph_close']);
     expect(tokens[1].children?.map((t) => t.type)).toEqual([
       'parsed_role_open',
       'role_body_open',
-      'heading_open',
```

Comment:
Hah, weird, the tests expected the unwanted inline parsing...

Reply:
Yeah, I think this was just documenting what it did without much thinking actually!